geoportailv3/static/js/search/searchdirective.js

```diff
@@ -0,0 +1,326 @@
+    templateUrl: appSearchTemplateUrl,
+    link:
+        /**
+         * @param {angular.Scope} scope Scope
+         * @param {angular.JQLite} element Element
+         * @param {angular.Attributes} attrs Atttributes
+         */
+        function(scope, element, attrs) {
+          // Empty the search field on focus and blur.
+          element.find('input').on('focus blur', function() {
+            $(this).val('');
+          });
+        }
+  };
+};
+
+
+app.module.directive('appSearch', app.searchDirective);
+
+
+
+/**
+ * @ngInject
+ * @constructor
+ * @param {angular.Scope} $scope Angular root scope.
+ * @param {app.Themes} appThemes Themes service.
+ * @param {angular.$compile} $compile Angular compile service.
+ * @param {ngeo.CreateGeoJSONBloodhound} ngeoCreateGeoJSONBloodhound The ngeo
+ *     create GeoJSON Bloodhound service.o
+ * @param {angularGettext.Catalog} gettextCatalog Gettext catalog
+ * @param {app.GetLayerForCatalogNode} appGetLayerForCatalogNode
+ * @param {app.ShowLayerinfo} appShowLayerinfo
+ * @export
+ */
+app.SearchDirectiveController =
+    function($scope, appThemes, $compile,
+        ngeoCreateGeoJSONBloodhound, gettextCatalog,
+        appGetLayerForCatalogNode, appShowLayerinfo) {
+  /**
+   * @type {Array}
+   */
+  this.layers = [];
+
+  /**
+   * @type {ol.FeatureOverlay}
+   * @private
+   */
+  this.featureOverlay_ = this.createFeatureOverlay_();
+
+  /**
+   * @type {app.GetLayerForCatalogNode}
+   * @private
+   */
+  this.getLayerFunc_ = appGetLayerForCatalogNode;
+
+  /**
+   * @type {app.ShowLayerinfo}
+   * @private
+   */
+  this.showLayerinfo_ = appShowLayerinfo;
+
+  /** @type {Bloodhound} */
+  var POIBloodhoundEngine = this.createAndInitPOIBloodhound_(
+      ngeoCreateGeoJSONBloodhound);
+  /** @type {Bloodhound} */
+  var LayerBloodhoundEngine = this.createAndInitLayerBloodhound_();
+
+  $scope.$on('gettextLanguageChanged', goog.bind(function(event, args) {
+    this.createLocalAllLayerData_(
+        appThemes, LayerBloodhoundEngine, gettextCatalog);
+  }, this));
+
+
+  /** @type {TypeaheadOptions} */
+  this['options'] = {
+    highlight: true
+  };
+
+  /** @type {Array.<TypeaheadDataset>} */
+  this['datasets'] = [{
+    source: POIBloodhoundEngine.ttAdapter(),
+    displayKey: function(suggestion) {
+      var feature = /** @type {ol.Feature} */ (suggestion);
+      return feature.get('label');
+    },
+    templates: {
+      header: function() {
+        return '<div class="header" translate>Addresses</div>';
+      },
+      suggestion: function(suggestion) {
+        var feature = /** @type {ol.Feature} */ (suggestion);
+
+        // A scope for the ng-click on the suggestion's « i » button.
+        var scope = $scope.$new(true);
+        scope['feature'] = feature;
+        scope['click'] = function(event) {
+          event.stopPropagation();
+        };
+
+        var html = '<p>' + feature.get('label') +
+            ' <span>(' + gettextCatalog.getString(
```

Comment:
Okay, after some reading https://angular-gettext.rocketeer.be/dev-guide/annotate-js/   it seems we can add *.js files to build to include js translations